models/projection/powder/mantid_projection_calculator.py

```diff
@@ -33,7 +40,13 @@ def calculate_projection(self, input_workspace, axis1, axis2):
                 str(dim0.getMaximum()) + ',' + str(dim0.getNBins())
             dim1 = dim1.getName() + ',' + str(dim1.getMinimum()) + ',' +\
                 str(dim1.getMaximum()) + ',' + str(dim1.getNBins())
-            return SliceMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, AlignedDim0=dim0,
-                           AlignedDim1=dim1)
+            SliceMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, AlignedDim0=dim0,
+                    AlignedDim1=dim1)
         else:
             raise NotImplementedError("Not implemented axis1 = %s and axis2 = %s" % (axis1, axis2))
+        # Now scale the energy axis if required - ConvertToMD always gives DeltaE in meV
+        if units == WAVENUMBER_LABEL:
+            TransformMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, Scaling=scale)
+            self._workspace_provider.get_workspace_handle(output_workspace).setComment('MSlice_in_wavenumber')
+        elif units != MEV_LABEL:
+            raise NotImplementedError("Unit %s not recognised. Only 'meV' and 'cm-1' implemented." % (units))
```

Comment:
Ah oops. Yes. Sorry, I should at a return value. I was only checking it with the GUI.